packages/router-component-store/src/lib/global-router-store/global-router-store-selectors.spec.ts

```diff
@@ -174,4 +191,24 @@ describe(`${GlobalRouterStore.name} selectors`, () => {
 
     await expect(firstValueFrom(routerStore.title$)).resolves.toBe('test-data');
   });
+
+  it('exposes a selector for specific router events', async () => {
+    const { navigateByUrl, routerStore } = await setup({
+      initialNavigation: false,
+    });
+    const expectedUrl =
+      '/login/kXpODMhMOluqn?ref=ngworker.github.io#test-fragment';
+    const whenNavigation = firstValueFrom(
+      routerStore
+        .selectRouterEvents(NavigationStart, NavigationEnd)
+        .pipe(take(2), toArray())
+    );
+
+    await navigateByUrl(expectedUrl);
+
+    await expect(whenNavigation).resolves.toEqual([
+      new NavigationStart(1, expectedUrl, 'imperative', null),
+      new NavigationEnd(1, expectedUrl, expectedUrl),
+    ]);
+  });
 });
```

Comment:
issue: Add tests covering a custom router event class.

Reply:
Not necessary. The base class `RouterEvent` isn't abastract but also isn't emitted by the Angular `Router`.